src/cr/cube/cubepart.py

```diff
@@ -754,6 +759,16 @@ def scale_std_err_row(self):
             return None
         return self.scale_std_dev_row / np.sqrt(self.columns_margin)
 
+    @lazyproperty
+    def selected_categories_label(self):
+        """Set of selected categories for all the dimensions."""
+        labels = []
+        for d in self._dimensions:
+            for s in d.selected_categories:
+                if s.get("name"):
+                    labels.append(s.get("name"))
+        return set(labels)
```

Comment:
I think this gets it done without the mutation:

```python
return frozenset(
    s["name"]
    for d in self.dimensions
    for s in d.selected_categories
    if s.get("name")
)

Reply:
Okay, a little more on this:

1. Name should be `selected_category_names`. It is a collection, so name is plural, not singular, and plural goes on _last_ word.
2. The return type should be `tuple` such that the order of the names is preserved. This also makes the return value immutable which is a must for lazyproperties.